### PR TITLE
fix vulnerabilities (gulp, gulp-mocha, gulp-mustache, gulp-if, del)

### PR DESCRIPTION
**Pull request by Farfurix**
Before:
```
found 30 vulnerabilities (8 low, 11 moderate, 10 high, 1 critical)
```
Current:
```
found 20 vulnerabilities (6 low, 11 moderate, 3 high)
```

### Changes:
1. Update `gulp`, `gulp-mocha`, `gulp-mustache`, `gulp-if`, `del`.
2. Use `gulp-step` for `gulp` tasks
3. Fix Gulpfile.js
4. Use `gulp-ll-next` instead of `gulp-ll`

### `npm install --save-dev del@3.0.0`
https://github.com/sindresorhus/del/releases/tag/v2.0.0:

> The API now returns a promise instead of accepting a callback function.

> ```diff
> And with gulp:
> 
>  var gulp = require('gulp');
>  var del = require('del');
> 
> - gulp.task('clean', function (cb) {
> -   del('unicorn.png', cb);
> + gulp.task('clean', function () {
> +   return del('unicorn.png');
>  });
> 
>  gulp.task('default', ['clean']);
> ```

### Gulp tool window
Before | After
-|-
gulp-commands-before | gulp-commands-after

### CODE REVIEW

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

File: Gulpfile.js [hunk 1]
```diff
@@ -105,25 +108,11 @@ function hang () {
...
-gulp.task('client-scripts', ['client-scripts-bundle'], () => {
-    return gulp.src('./src/client/index.js.wrapper.mustache')
-        .pipe(mustache({ source: fs.readFileSync('./lib/client/hammerhead.js').toString() }))
-        .pipe(rename('hammerhead.js'))
-        .pipe(gulpif(!util.env.dev, uglify()))
-        .pipe(gulp.dest('./lib/client'));
-});
-
-gulp.task('client-scripts-bundle', ['clean'], () => {
+gulp.step('client-scripts-bundle', () => {
```
Comment (miherlosev) [inline Gulpfile.js]: I guess we can migrate to gulp 4 using a more simple manner
```js
//Gulp 3
gulp.task('client-scripts', ['client-scripts-bundle'], () => {
});

//Gulp 4
gulp.task('client-scripts', gulp.series(['client-scripts-bundle'], () => {
})); 
```

See https://github.com/gulpjs/gulp/blob/4.0/docs/API.md#gulpseriestasks

Check it and discuss with @AndreyBelym

Comment (miherlosev) [reply Gulpfile.js]: After discussing with @AndreyBelym we decide to use `gulp-step`.

File: Gulpfile.js [hunk 2]
```diff
@@ -148,12 +138,28 @@ gulp.task('client-scripts-bundle', ['clean'], () => {
         .pipe(gulp.dest('./lib/client'));
 });
 
-gulp.task('server-scripts', ['clean'], () => {
+gulp.step('client-scripts-render', () => {
```
Comment (miherlosev) [inline Gulpfile.js]: rename to `gulp.step('client-scripts-processing')`

File: Gulpfile.js [hunk 3]
```diff
@@ -167,10 +173,20 @@ gulp.task('lint', () => {
...
+            'server-scripts',
+            'templates',
+            'lint'
+        )
+    )
+);
 
 // Test
-gulp.task('test-server', ['build'], () => {
+gulp.step('test-server-run', () => {
```
Comment (miherlosev) [inline Gulpfile.js]: gulp.step('mocha')

File: Gulpfile.js [hunk 4]
```diff
@@ -179,29 +195,42 @@ gulp.task('test-server', ['build'], () => {
         }));
 });
 
-gulp.task('test-client', ['build'], () => {
-    gulp.watch('./src/**', ['build']);
+gulp.task('test-server', gulp.series('build', 'test-server-run'));
+
+gulp.step('test-client-run', () => {
```
Comment (miherlosev) [inline Gulpfile.js]: `gulp.step('qunit', ()`

File: Gulpfile.js [hunk 5]
```diff
@@ -179,29 +195,42 @@ gulp.task('test-server', ['build'], () => {
...
+
+gulp.step('set-dev-mode', done => {
+    util.env.dev = true;
+    done();
+});
+
+gulp.task('test-client-dev', gulp.series('set-dev-mode', 'test-client'));
 
-gulp.task('test-client-travis', ['build'], () => {
+gulp.step('test-client-travis-run', () => {
```
Comment (miherlosev) [inline Gulpfile.js]: `gulp.step('travis-saucelabs-qunit'`

File: Gulpfile.js [hunk 6]
```diff
@@ -179,29 +195,42 @@ gulp.task('test-server', ['build'], () => {
...
     return gulp
         .src('./test/client/fixtures/**/*-test.js')
         .pipe(qunitHarness(CLIENT_TESTS_SETTINGS, SAUCELABS_SETTINGS));
 });
 
-gulp.task('http-playground', ['set-dev-mode', 'build'], () => {
+gulp.task('test-client-travis', gulp.series('build', 'test-client-travis-run'));
+
+gulp.step('http-playground-run', () => {
```
Comment (miherlosev) [inline Gulpfile.js]: `gulp.step('http-playground-server'`

File: Gulpfile.js [hunk 7]
```diff
@@ -179,29 +195,42 @@ gulp.task('test-server', ['build'], () => {
...
     require('./test/playground/server.js').start();
 
     return hang();
 });
 
-gulp.task('https-playground', ['set-dev-mode', 'build'], () => {
+gulp.task('http-playground', gulp.series('set-dev-mode', 'build', 'http-playground-run'));
+
+gulp.step('https-playground-run', () => {
```
Comment (miherlosev) [inline Gulpfile.js]: gulp.step('https-playground-server'

**Comment by Farfurix** [comment]
@testcafe-build-bot retest

**Pull request merged**